src/chi-vue/src/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,338 @@
+import { Component, Prop, Vue } from 'vue-property-decorator';
+import { uuid4 } from '../../utils/utils';
+import {
+  activeClass,
+  inverseClass,
+  buttonClass,
+  buttonContentClass,
+  buttonGroupClass,
+  iconButtonClass,
+  lightClass,
+  buttonFlatClass,
+  disabledClass,
+  iconClass,
+  paginationClass,
+  paginationResultsClass,
+  paginationCompactClass,
+  paginationContentClass,
+  paginationLabelClass,
+  paginationPageSizeCLass,
+  paginationStartClass,
+  paginationCenterClass,
+  inputClass,
+  paginationJumperClass,
+  paginationEndClass
+} from '../../constants/classes';
+import { pageChangeEvent, pageSizeChangeEvent } from '../../constants/events';
+
+@Component
+export default class Pagination extends Vue {
+  @Prop() pages!: number;
+  @Prop() firstLast!: boolean;
+  @Prop() inverse!: boolean;
+  @Prop() currentPage!: number;
+  @Prop() results!: number;
+  @Prop() pageSize!: number;
+  @Prop() pageJumper!: number;
+  @Prop() compact!: boolean;
+  @Prop() ariaLabel!: boolean;
+  @Prop() size!: string;
+
+  _pageJumperUuid!: string;
+  _distanceFromCurrent!: number;
+  _lastRenderedPage!: number;
+  _pagesToRender: any = [];
+  _startPage!: JSX.Element | null;
+  _lastPage!: JSX.Element | null;
+  _results: any;
```

Comment:
Could these _anys_ be replaced by some kind of custom type?

Reply:
+1